js/apps/admin-ui/src/user/UserForm.tsx

```diff
@@ -328,9 +328,10 @@ export const UserForm = ({
           <Button
             data-testid={!user?.id ? "create-user" : "save-user"}
             isDisabled={
```

Comment:
Another possibility here would be to import `isSubmittable` and re-use it. Or perhaps a better solution would be to allow passing in `isDisabled` to `FormSubmitButton` and use it as a fallback to determine the disabled state of the button.

Reply:
Thank you for the suggestion, I have modified `FormSubmitButtonProps` to support two usages:
* If `formState` is set, `isDisabled` will be invalid - this will use the default `isSubmittable` behaviour
* If `isDisabled` is set, `formState` will be invalid - as they are providing their own `isDisabled` computation, there's no need to provide `formState`.

```
export type FormSubmitButtonProps = Omit<ButtonProps, "isDisabled"> &
  (
    | {
        formState: FormState<FieldValues>;
        allowNonDirty?: boolean;
        allowInvalid?: boolean;
        isDisabled?: never;
      }
    | {
        formState?: never;
        allowNonDirty?: never;
        allowInvalid?: never;
        isDisabled: boolean;
      }
  );
```

Reply:
What do you think @jonkoops ? Or you prefer the alternative:

**Allowing `isDisabled` to be passed**
```js
// FormSubmitButton.tsx
import { Button, ButtonProps } from "@patternfly/react-core";
import { PropsWithChildren } from "react";
import { FieldValues, FormState } from "react-hook-form";

export type FormSubmitButtonProps = Omit<ButtonProps, "isDisabled"> &
{
  formState: FormState<FieldValues>;
  allowNonDirty?: boolean;
  allowInvalid?: boolean;
  isDisabled?: boolean;
};

const isSubmittable = (
  formState: FormState<FieldValues>,
  allowNonDirty: boolean,
  allowInvalid: boolean,
) => {
  return (
    (formState.isValid || allowInvalid) &&
    (formState.isDirty || allowNonDirty) &&
    !formState.isLoading &&
    !formState.isValidating &&
    !formState.isSubmitting
  );
};

export const FormSubmitButton = ({
  formState,
  isDisabled = false,
  allowInvalid = false,
  allowNonDirty = false,
  children,
  ...rest
}: PropsWithChildren<FormSubmitButtonProps>) => {
  return (
    <Button
      variant="primary"
      isDisabled={
        (formState && !isSubmittable(formState, allowNonDirty, allowInvalid)) || isDisabled
      }
      {...rest}
      type="submit"
    >
      {children}
    </Button>
  );
};

```

Reply:
@jhchong92 let's go for the latter suggestion, I think `isDisabled` should be considered an override that could occasionally be passed if needed.